File: 2024-2/0682/unidad06.d_Web_04/SoftPub/SoftPubGrupoControler/DAO/GrupoDAO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubGrupoModel;

namespace SoftPubGrupoControler.DAO
{
    public interface GrupoDAO
    {        
        int? insertar(Grupo grupo);

        BindingList<Object> listarTodos();

        BindingList<Object> listarGruposConProduccion();

        BindingList<Object> buscarGrupos(string nombre);

        Grupo obtenerPorId(int idGrupo);
    }
}

File: 2024-2/Proyecto DAO SoftPub Java/SoftPubGrupoDA/src/pe/edu/pucp/softpub/grupo/dao/GrupoDAO.java
package pe.edu.pucp.softpub.grupo.dao;

import java.util.ArrayList;
import pe.edu.pucp.softpub.grupo.model.Grupo;

public interface GrupoDAO {

    public Integer insertar(Grupo grupo);    

    public Integer modificar(Grupo grupo);

    public Integer eliminar(Grupo grupo);

    public ArrayList<Grupo> listarTodos();

    public Grupo obtenerPorId(Integer idGrupo);
    
}
